app/utils/img_processing/tesseract_api.py

```diff
@@ -3,76 +3,164 @@
+        data_dict = image_to_data(
+            image, lang=self.lang, config=TESS_CONFIG, output_type='dict')
+        return data_dict
 
+    def extract_text(self, image):
+        """
+        Arguments:
+        Single full-page image
+
+        Actions:
+        Extract text from image with Tesseract OCR
+        
+        Returns:
+        Transcribed text (string)
         """
 
-        # Convert image data bytestring to image object
-        nparr = np.fromstring(document, np.uint8)
-        img = imdecode(nparr, IMREAD_COLOR)
+        text = image_to_string(
+            image, lang=self.lang, config=TESS_CONFIG)
 
-        img_blocks = self.img_preprocess(img)
-        flagged = False
+        return text
 
-        transcribed_text = []
+    def word_moderation(self, data_dict):
+        """
+        Arguments:
+        Data dictionary generated by Tesseract from extract_data method 
+
+        Actions:
+        Moderate content by word
+        
+        Returns:
+        Bad content flag (T/F)
+        """
 
-        for img_block in img_blocks:
-            ttext = image_to_string(
-                img_block, lang=self.lang, config=TESS_CONFIG)
+        content_flagged = False
+        for word in data_dict['text']:
+            if self.text_moderator.check_word(str(word).lower()):
+                content_flagged = True
+        
+        return content_flagged
 
-            for word in ttext:
-                # check moderation status of word in paragraph
-                if self.text_moderator.check_word(str(word).lower()):
-                    flagged = True
+    def confidence_flag(self, data_dict):
+        """
+        Arguments:
+        Data dictionary generated by Tesseract from extract_data method 
+
+        Actions:
+        Calculate mean confidence for entire page
+        
+        Returns:
+        Low confidence flag (T/F)
+        """
+
+        confidence_list = list(map(int, data_dict['conf']))
+        page_confidence = sum(confidence_list) / len(confidence_list) / 100
 
-            transcribed_text.append(ttext)
+        return (page_confidence < 0.85)
 
-        return False, flagged, ' '.join(transcribed_text)
+    def transcribe(self, image):
+        """
+        Arguments:
+        Single full-page image
+
+        Actions:
+        1. Preprocesses image
+        2. Extract text
+        3. Moderate content
+        4. Get confidence
+
+        Returns:
+        1. Transcribed text (string)
+        2. Bad content flag (T/F)
+        3. Low confidence flag (T/F)
+        """
+        # Preprocess image
+        features = self.img_preprocess(image)
+        # Extract text
+        text = self.extract_text(features)
+        # Extract data
+        data_dict = self.extract_data(features)
+        # Moderate content
+        content_flagged = self.word_moderation(data_dict)
+        # Get confidence
+        low_confidence = self.confidence_flag(data_dict)
```

Comment:
simarly these probably don't need comments if the function names explain what's happening

- `img_preprocess` -> `img_to_features`
- `self.word_moderation` -> `self.detect_innappropriate_content`
- `self.confidence_flag` -> `self.is_confident`

Reply:
Comments have been removed